zerver/migrations/0574_backfill_directmessagegroup_group_size.py

```diff
@@ -0,0 +1,72 @@
+# Generated by Django 5.0.6 on 2024-08-16 06:44
+
+from django.db import migrations, transaction
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import Count, Model, OuterRef, Subquery
+
+BATCH_SIZE = 1000
+
+
+def backfill_group_size_field_for_direct_message_groups(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    direct_message_group_model = apps.get_model("zerver", "DirectMessageGroup")
+    recipient_model = apps.get_model("zerver", "Recipient")
+    subscription_model = apps.get_model("zerver", "Subscription")
+
+    DIRECT_MESSAGE_GROUP = 3
+
+    direct_message_group_recipient_entries = recipient_model.objects.filter(
+        type=DIRECT_MESSAGE_GROUP
```

Comment:
I don't know the exact reason, but somehow I am not able to use `Recipient.DIRECT_MESSAGE_GROUP` over here.

I used this because I found a similar representation in [`0419_backfill_message_realm.py`](https://github.com/zulip/zulip/blob/6cc480e2b5df1e96848764125b896b135ed0cb09/zerver/migrations/0419_backfill_message_realm.py#L11):
```py
    RECIPIENT_PERSONAL = 1
    RECIPIENT_STREAM = 2
    RECIPIENT_HUDDLE = 3
```

Reply:
Yep, migration files can't depend on any code outside the migration, and that includes non-database values set in `models` files.

To understand why, imagine what would happen if we swapped around those numbers in a next migration: We wouldn't want the behavior of this migration to be impacted!